models/CGCNNStack.py

```diff
@@ -0,0 +1,64 @@
+import torch
+import torch.nn.functional as F
+from torch.nn import ModuleList
+from torch_geometric.nn import CGConv, BatchNorm, global_mean_pool
+from .Base import Base
+
+
+class CGCNNStack(Base):
+    def __init__(
+        self,
+        input_dim: int,
+        output_dim: list,
+        output_type: list,
+        num_nodes: int,
+        config_heads: {},
+        edge_dim: int = 0,
+        dropout: float = 0.25,
+        num_conv_layers: int = 16,
+        ilossweights_hyperp: int = 1,  # if =1, considering weighted losses for different tasks and treat the weights as hyper parameters
+        loss_weights: list = [1.0, 1.0, 1.0],  # weights for losses of different tasks
+        ilossweights_nll: int = 0,  # if =1, using the scalar uncertainty as weights, as in paper
+        # https://openaccess.thecvf.com/content_cvpr_2018/papers/Kendall_Multi-Task_Learning_Using_CVPR_2018_paper.pdf
+    ):
+        super().__init__()
+
+        # self.hidden_dim = hidden_dim
+        self.hidden_dim = input_dim
+        self.dropout = dropout
+        self.num_conv_layers = num_conv_layers
+        self.convs = ModuleList()
+        self.batch_norms = ModuleList()
+        self.convs.append(
+            CGConv(
+                channels=input_dim,
```

Comment:
`channels` looks like a tuple of input and output - I think they just combined into one input rather than separate

Reply:
I doubt it is what we want. It says "A tuple corresponds to the sizes of source and target dimensionalities." I am thinking it means for source and target nodes. The option is there in other models also.

Reply:
Usually, channels is the number of neuron (or channels, or filters) in the hidden layer.

Reply:
Sure, that's what we were talking about. There is only one channel parameter which means the input and output channels have to be the same in this model, unless I missed something.

Reply:
If channels is passed as an integer, that means that the input channels and output channels are the same only for that specific layer, not for the entire model.